src/plugins/intel_cpu/src/transformations/cpu_opset/common/pass/rope_fusion.cpp

```diff
@@ -746,6 +746,34 @@ ov::intel_cpu::RoPEFusionQwen::RoPEFusionQwen(int split_output_id) {
             config.slice_stop = config.slice_start + config.head_cnt * config.head_size;
         }
 
+        if (pattern_map.count(reshape_special)) {
+            // check reshape_special shape correctness
+            auto reshape_special_node = pattern_map.at(reshape_special).get_node_shared_ptr();
+            auto data_shape = reshape_special_node->get_input_partial_shape(0);
+            auto reshape_shape = pattern_map.at(const_shape);
+            auto check_const = [&data_shape](Output<Node> output) -> bool {
+                auto node = std::dynamic_pointer_cast<opset1::Constant>(output.get_node_shared_ptr());
+                const auto& target = node->cast_vector<int32_t>();
+                // ensure data_shape and target_shape have correct rank
+                if (target.size() < 3 || data_shape.rank().is_dynamic() || (data_shape.rank().is_static() && data_shape.rank().get_length() < 2)) {
+                    return false;
+                }
+                if (data_shape[data_shape.size() - 1].is_static() && data_shape[data_shape.size() - 2].is_static()) {
+                    auto head_size = data_shape[data_shape.size() - 1].get_length();
+                    auto head_cnt = data_shape[data_shape.size() - 2].get_length();
```

Comment:
`RoPENode::Config` [already has](https://github.com/openvinotoolkit/openvino/pull/24727/files#diff-8c8905f6745d31f66e0f0dfae7c5b5d614fd03c3b643f0f29250f24879b2a9f4R735-R736) the filled `head_cnt` and `head_size` values. Maybe we can reuse them in this reshape check?